### PR TITLE
add a factory method for creating hdfs env

### PR DESCRIPTION
**Pull request by shuzhang1989**
There is no way we can use HDFSEnv even if we compile with HDFS and install as shared_lib.
implementing a factory method that creates the HDFS environment.

### CODE REVIEW

File: util/env_hdfs.cc [hunk 1]
```diff
@@ -594,6 +594,11 @@ Status HdfsEnv::NewLogger(const std::string& fname,
   return Status::OK();
 }
 
+	// The factory method for creating an HDFS Env
+Status NewHdfsEnv(Env** hdfs_env, const std::string& fsname) {
+	*hdfs_env = new HdfsEnv(fsname);
+	return Status::OK();
```
Comment (siying) [inline util/env_hdfs.cc]: Can you follow the indenting convention?

**Comment by shuzhang1989** [comment]
@siying formatted.

File: util/env_hdfs.cc [hunk 2]
```diff
@@ -607,6 +612,11 @@ namespace rocksdb {
                                    const EnvOptions& options) {
    return Status::NotSupported("Not compiled with hdfs support");
  }
+
+ Status NewHdfsEnv(Env** hdfs_env, const std::string& fsname) {
+	 return Status::NotSupported("Not compiled with hdfs support");
```
Comment (siying) [inline util/env_hdfs.cc]: Indenting?

**Comment by shuzhang1989** [comment]
@siying formatted again to remove indents and additional blank lines. The building errors looks not related to this change, can you confirm? Thanks! happy new year!

**Pull request merged**